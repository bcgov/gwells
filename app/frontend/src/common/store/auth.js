export const SET_KEYCLOAK = 'SET_KEYCLOAK'

const auth = {
  state: {
    keycloak: {}
  },
  mutations: {
    [SET_KEYCLOAK] (state, payload) {
      state.keycloak = payload
    }
  },
  getters: {
    keycloak (state) {
      return state.keycloak
    },
    userRoles (state) {
      if (state.keycloak && state.keycloak.authenticated) {
        return {
          registry: {
            // map SSO roles to web app permissions
            view: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority') ||
              state.keycloak.hasRealmRole('registries_viewer') ||
              state.keycloak.hasRealmRole('registries_adjudicator')),
            edit: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority') ||
              state.keycloak.hasRealmRole('registries_adjudicator')),
            approve: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority'))
          },
          wells: {
            view: (!!state.keycloak.hasRealmRole('gwells_admin')),
            edit: (!!state.keycloak.hasRealmRole('gwells_admin')),
            approve: (!!state.keycloak.hasRealmRole('gwells_admin'))
          },
          submissions: {
            view: (!!state.keycloak.hasRealmRole('gwells_admin')),
            edit: (!!state.keycloak.hasRealmRole('gwells_admin')),
            approve: (!!state.keycloak.hasRealmRole('gwells_admin'))
          }
        }
      } else {
        return {
          registry: {},
          wells: {},
          submissions: {}
        }
      }
    }
  }
}

export default auth
