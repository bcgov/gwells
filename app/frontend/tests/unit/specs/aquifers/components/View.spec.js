/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/* eslint-disable import/first */

import '../../../mocks/mapbox-gl'
import { mount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import axios from 'axios'
import { cloneDeep, merge, omit } from 'lodash'

import auth from '@/common/store/auth.js'
import aquiferStore from '@/aquifers/store/index.js'
import documentState from '@/common/store/documents.js'
import ViewComponent from '@/aquifers/components/View.vue'

jest.mock('axios')

const localVue = createLocalVue()
localVue.use(Vuex)

const aquiferFixture = {
  aquifer_id: '4',
  aquifer_name: 'Aggasiz',
  location_description: 'Aggasiz/Sea Bird Island',
  material_description: 'Sand and Gravel',
  litho_stratographic_unit: 'Fraser River Sediments',
  subtype_description: 'Unconfined sand and gravel - large river system',
  vulnerability_description: 'High',
  area: '63.3',
  productivity_description: 'High',
  demand: 'Low',
  mapping_year: '1993',
  resources: [],
  licence_details: {
    licence_count: 4,
    licences_updated: {
      update_date__max: '2019-11-04T19:47:48.075000Z'
    },
    num_wells_with_ems: 3,
    num_artesian_wells: 4,
    wells_updated: {
      update_date__max: '2016-02-09T21:55:31Z'
    },
    num_wells: 4,
    obs_wells: [{
      well_tag_number: 117042,
      observation_well_number: '2',
      observation_well_status: 'Active'
    }, {
      well_tag_number: 100125,
      observation_well_number: '2',
      observation_well_status: 'Active'
    }],
    wells_by_licence: [{
      licence_number: 501525,
      well_tag_numbers_in_licence: [117041]
    }, {
      licence_number: 501526,
      well_tag_numbers_in_licence: [117043]
    }, {
      licence_number: 501527,
      well_tag_numbers_in_licence: [117044]
    }, {
      licence_number: 501528,
      well_tag_numbers_in_licence: [117045]
    }],
    usage: [{
      purpose__description: '03A - Irrigation: Local Provider ',
      total_qty: 2730000.0
    }, {
      purpose__description: 'WSA09 - Processing & Manufacturing ',
      total_qty: 980000.0
    }, {
      purpose__description: '03B - Irrigation: Private ',
      total_qty: 6850000.0
    }],
    lic_qty: [{
      purpose__description: '03A - Irrigation: Local Provider ',
      total_qty: 1
    }, {
      purpose__description: 'WSA09 - Processing & Manufacturing ',
      total_qty: 1
    }, {
      purpose__description: '03B - Irrigation: Private ',
      total_qty: 2
    }],
    hydraulically_connected: true,
    aquifer_notations: true
  }
}

describe('View Component', () => {
  let fetch = null

  const component = (options, storeState = {}) => {
    const store = new Vuex.Store({
      modules: { auth, aquiferStore, documentState }
    })
    store.replaceState(merge(cloneDeep(store.state), cloneDeep(storeState)))

    return mount(ViewComponent, {
      localVue,
      store,
      mocks: {
        $router: {
          push: jest.fn(),
          replace: jest.fn()
        },
        $route: {
          path: `/aquifers/${aquiferFixture.aquifer_id}`,
          params: {
            id: aquiferFixture.aquifer_id
          }
        }
      },
      stubs: ['router-link', 'aquifer-documents', 'aquifer-form', 'b-popover', 'pie-chart', 'single-aquifer-map'],
      methods: {
        fetch,
        fetchAquifer: jest.fn(),
        fetchWells: jest.fn(),
        fetchFiles: jest.fn(),
        navigateToView: jest.fn()
      },
      ...options
    })
  }

  beforeEach(() => {
    fetch = jest.fn()
    axios.get.mockResolvedValue({})
  })

  it('queries aquifer on load', async () => {
    component()

    await Vue.nextTick()

    expect(fetch).toHaveBeenCalled()
  })

  describe('View mode', () => {
    it('matches the snapshot', () => {
      const wrapper = component({
        propsData: { edit: false }
      }, {
        aquiferStore: {
          view: {
            record: aquiferFixture
          }
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Edit mode', () => {
    it('matches the snapshot', () => {
      const wrapper = component({
        propsData: { edit: true }
      }, {
        aquiferStore: {
          view: {
            record: aquiferFixture
          }
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    describe('On save', () => {
      it('resets showSaveSuccess to false', () => {
        const wrapper = component({
          data () {
            return {
              showSaveSuccess: true
            }
          }
        }, {
          aquiferStore: {
            view: {
              record: aquiferFixture
            }
          }
        })

        axios.patch.mockResolvedValue(true)

        expect(wrapper.vm.showSaveSuccess).toBe(true)
        wrapper.vm.save()
        expect(wrapper.vm.showSaveSuccess).toBe(false)
      })

      it('resets fieldErrors to empty object', () => {
        const wrapper = component({
          data () {
            return {
              fieldErrors: { a: ['1'] }
            }
          },
          propsData: { edit: true }
        }, {
          aquiferStore: {
            view: {
              record: aquiferFixture
            }
          }
        })
        axios.patch.mockResolvedValue(true)

        wrapper.vm.save()
        expect(wrapper.vm.fieldErrors).toEqual({})
      })

      it('sends a patch with the contents of record on save', () => {
        const wrapper = component({
          computed: {
            id () {
              return 10
            }
          },
          data () {
            return {
              form: aquiferFixture
            }
          }
        }, {
          aquiferStore: {
            view: {
              record: aquiferFixture
            }
          }
        })

        axios.patch.mockResolvedValue(true)

        wrapper.vm.save()

        expect(axios.patch).toHaveBeenCalledWith('aquifers/10', omit(aquiferFixture, 'area', 'licence_details'))
      })
    })
  })
})
