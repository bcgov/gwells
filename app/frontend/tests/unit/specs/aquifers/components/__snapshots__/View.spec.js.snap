// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Component Edit mode matches the snapshot 1`] = `
<div
  id="aquifer-detail"
>
  <div
    class="card mb-3 container d-print-none"
  >
    <!---->
    <!---->
    <ol
      class="breadcrumb py-0 my-2"
    >
      <li
        class="breadcrumb-item"
      >
        <router-link-stub
          class=""
          target="_self"
        >
          Aquifer Search
        </router-link-stub>
      </li>
      <li
        class="breadcrumb-item"
      >
        <router-link-stub
          class=""
          target="_self"
        >
          Aquifer 4 Summary
        </router-link-stub>
      </li>
      <li
        class="breadcrumb-item active"
      >
        <span
          aria-current="location"
        >
          Edit Aquifer
        </span>
      </li>
    </ol>
    <!---->
    <!---->
  </div>
   
  <div>
    <div
      class="card container container-wide card-container p-0 pb-5 main-card p-4"
    >
      <!---->
      <!---->
      <div
        class="card-body"
      >
        <!---->
        <!---->
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <div>
          <div
            id="form-loading-spinner"
          >
            <div
              class="fa-2x text-center"
            >
              <i
                class="fa fa-circle-o-notch fa-spin"
              />
            </div>
          </div>
           
          <div
            class="row border-bottom mb-3 pb-2"
          >
            <div
              class="col"
            >
              <h4>
                Aquifer 4 Summary - Edit
              </h4>
            </div>
          </div>
           
          <aquifer-form-stub
            fielderrors="[object Object]"
            files="[object Object]"
            loadingdata="true"
            record="[object Object]"
            showid=""
          />
           
          <div
            class="card mt-5"
          >
            <div
              class="card-body"
            >
              <h6
                class="card-title mb-0"
                id="changeHistoryTitle"
              >
                Change History
      
                <span
                  class="ml-3"
                >
                  <button
                    class="btn btn-outline-primary btn-sm"
                    link=""
                    type="button"
                  >
                    Show
                  </button>
                </span>
              </h6>
               
              <div
                id="historyList"
              >
                <div>
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`View Component View mode matches the snapshot 1`] = `
<div
  id="aquifer-detail"
>
  <div
    class="card mb-3 container d-print-none"
  >
    <!---->
    <!---->
    <ol
      class="breadcrumb py-0 my-2"
    >
      <li
        class="breadcrumb-item"
      >
        <router-link-stub
          class=""
          target="_self"
        >
          Aquifer Search
        </router-link-stub>
      </li>
      <li
        class="breadcrumb-item active"
      >
        <span
          aria-current="location"
        >
          Aquifer Summary
        </span>
      </li>
    </ol>
    <!---->
    <!---->
  </div>
   
  <div>
    <div
      class="card container container-wide card-container p-0 pb-5 main-card"
    >
      <!---->
      <!---->
      <div
        class="card-body"
      >
        <!---->
        <!---->
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <!---->
         
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="aquifer-detail col-md-12 col-lg-5 col-12"
            >
              <div
                class="row"
              >
                <div
                  class="pt-0 pl-4 pb-4 pr-4 col"
                >
                  <div
                    class="d-flex justify-content-between align-items-center"
                  >
                    <h4
                      class="color-grey main-title mt-4"
                    >
                      Aquifer 4 Summary
                    </h4>
                     
                    <div>
                      <!---->
                       
                      <a
                        class="ml-2 print-button fa fa-print fa-lg d-print-none"
                        href="#"
                        title="Print"
                      />
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="row"
              >
                <div
                  class="pl-4 pr-4 aquifer-main-information-list col-sm-12 col-12"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Aquifer number
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                      id="aquifer-view-number"
                    >
                      4
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Year of mapping
                    </div>
                     
                    <div
                      class="aquifer-mapping-year col-md-3 col-lg-6 col-6"
                    >
                      1993
                    </div>
                  </div>
                   
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Aquifer name
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                      id="aquifer-view-name"
                    >
                      Aggasiz
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Litho stratigraphic unit
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Fraser River Sediments
                    </div>
                  </div>
                   
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Descriptive location
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Aggasiz/Sea Bird Island
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Vulnerability
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    </div>
                  </div>
                   
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Material type
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Subtype
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    </div>
                  </div>
                   
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Quality concerns
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Productivity
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    </div>
                  </div>
                   
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Size (kmÂ²)
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      63.3
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      
                    Calculated well density
                    
                      <i
                        class="fa fa-question-circle color-info fa-xs pt-0 mt-0 d-print-none"
                        id="calculated-demand-density-info"
                        tabindex="0"
                      />
                       
                      <b-popover-stub
                        content="Calculated based on the number of wells known to be completed in the aquifer per square kilometer. Light â¤ 4 kmÂ²; Medium 4 â 20 kmÂ²; Heavy > 20 kmÂ²"
                        target="calculated-demand-density-info"
                        triggers="hover focus"
                      />
                    </div>
                     
                    <div
                      class="col-md-3 col-lg-6 col-6"
                    >
                      Low
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="p-0 col-md-12 col-lg-7 col-12"
              id="map-container"
            >
              <!---->
               
              <single-aquifer-map-stub
                aquiferid="4"
              />
            </div>
          </div>
           
          <div
            class="row mt-5 aquifer-details"
          >
            <div
              class="col-lg-6 col-xl-4 col-12"
            >
              <h5
                class="mt-3 border-bottom pb-4 main-title"
              >
                Well Information
              </h5>
               
              <ul
                class="ml-0 mr-0 mt-4 mb-0 p-0 aquifer-information-list"
              >
                <div
                  class="aquifer-information-list-divider"
                />
                 
                <li>
                  <dt>
                    Number of wells correlated to the aquifer
                  </dt>
                   
                  <dd
                    class="m-0"
                  >
                    <router-link-stub
                      to="[object Object]"
                    >
                      
                    4
                  
                    </router-link-stub>
                  </dd>
                </li>
                 
                <li>
                  <dt>
                    
                  Number of uncorrelated wells within mapped aquifer extent
                  
                    <i
                      class="fa fa-question-circle color-info fa-xs pt-0 mt-0 d-print-none"
                      id="uncorrelated-wells-count"
                      tabindex="0"
                    />
                     
                    <b-popover-stub
                      content="The total number of wells that fall within the aquifer polygon but have not been correlated to any aquifer. These wells may potentially be located within this aquifer or may be completed within an aquitard or within another aquifer located at a different depth."
                      target="uncorrelated-wells-count"
                      triggers="hover focus"
                    />
                  </dt>
                   
                  <dd
                    class="m-0"
                  >
                    
                  0
                
                  </dd>
                </li>
                 
                <li>
                  <dt>
                    Artesian wells
                  </dt>
                   
                  <dd
                    class="m-0"
                  >
                    <router-link-stub
                      to="[object Object]"
                    >
                      
                    4 artesian wells in aquifer
                  
                    </router-link-stub>
                  </dd>
                </li>
              </ul>
               
              <p>
                <i>
                  Well info last updated 2/9/2016
                </i>
              </p>
               
              <h5
                class="mt-5 border-bottom pb-4 main-title"
              >
                Documentation
              </h5>
               
              <aquifer-documents-stub
                files="[object Object]"
                highlightregexp="/[_ -]factsheet[_ -]*.*\\\\.pdf$/i"
                highlighttitle="Factsheets"
                id="4"
                publicfilestitle="Other Documents"
              />
            </div>
             
            <div
              class="col-lg-6 col-xl-4 col-12"
            >
              <h5
                class="mt-3 border-bottom pb-4 main-title"
              >
                Licensing Information
              </h5>
               
              <div>
                <p>
                  
                The licensing summaries should be considered estimates. Due to complexities in the structure
                of the licensing data, reported values should be confirmed through the
                
                  <a
                    class="d-print-url"
                    href="https://j200.gov.bc.ca/pub/ams/Default.aspx?PossePresentation=AMSPublic&PosseObjectDef=o_ATIS_DocumentSearch&PosseMenuName=WS_Main"
                    target="_blank"
                  >
                    
                  eâlicensing portal
                  </a>
                  .
              
                </p>
              </div>
               
              <ul
                class="ml-0 mr-0 mt-4 mb-0 p-0 aquifer-information-list"
              >
                <div
                  class="aquifer-information-list-divider"
                />
                 
                <li>
                  <dt>
                    Number of groundwater licences
                  </dt>
                   
                  <dd
                    class="m-0"
                  >
                    4
                  </dd>
                </li>
                 
                <li>
                  <dt>
                    Water withdrawal volume (annual)
                  </dt>
                   
                  <dd
                    class="m-0"
                  >
                    No information available.
                  </dd>
                </li>
              </ul>
               
              <div>
                <div
                  class="row pt-5"
                >
                  <div
                    class="pb-5 col-md-6 col-lg-12 col-12"
                  >
                    <h5
                      class="pie-chart-title"
                    >
                      Licensed volume by purpose (millions of cubic meters)
                    </h5>
                     
                    <pie-chart-stub
                      chartid="pie-chart"
                      chartoptions="[object Object]"
                      class="mt-3"
                      cssclasses=""
                      data="2.73,0.98,6.85"
                      height="400"
                      labels="Irrigation: Local Provider  2.73,Processing & Manufacturing  0.98,Irrigation: Private  6.85"
                      plugins=""
                      width="400"
                    />
                  </div>
                   
                  <div
                    class="pb-5 col-md-6 col-lg-12 col-12"
                  >
                    <h5
                      class="pie-chart-title"
                    >
                      Number of licences by purpose
                    </h5>
                     
                    <pie-chart-stub
                      chartid="pie-chart"
                      chartoptions="[object Object]"
                      class="mt-3"
                      cssclasses=""
                      data="1,1,2"
                      height="400"
                      labels="Irrigation: Local Provider ,Processing & Manufacturing ,Irrigation: Private "
                      plugins=""
                      width="400"
                    />
                  </div>
                </div>
              </div>
               
              <table
                aria-busy="false"
                aria-colcount="2"
                class="table b-table table-striped"
                id="licenses"
              >
                <!---->
                <!---->
                <thead
                  class=""
                  role="rowgroup"
                >
                  <!---->
                  <tr
                    role="row"
                  >
                    <th
                      aria-colindex="1"
                      class=""
                      role="columnheader"
                      scope="col"
                    >
                      Licence Number
                    </th>
                    <th
                      aria-colindex="2"
                      class=""
                      role="columnheader"
                      scope="col"
                    >
                      Well Tag Numbers In Licence
                    </th>
                  </tr>
                </thead>
                <!---->
                <tbody
                  class=""
                  role="rowgroup"
                >
                  <!---->
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      class=""
                      role="cell"
                    >
                      <a
                        href="https://j200.gov.bc.ca/pub/ams/Default.aspx?PossePresentation=AMSPublic&PosseObjectDef=o_ATIS_DocumentSearch&PosseMenuName=WS_Main&Criteria_LicenceNumber=501525"
                        target="_blank"
                      >
                        
                  501525
                
                      </a>
                    </td>
                    <td
                      aria-colindex="2"
                      class=""
                      role="cell"
                    >
                      <ul
                        class="p-0 m-0"
                      >
                        <li>
                          <router-link-stub
                            to="[object Object]"
                          >
                            117041
                          </router-link-stub>
                        </li>
                      </ul>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      class=""
                      role="cell"
                    >
                      <a
                        href="https://j200.gov.bc.ca/pub/ams/Default.aspx?PossePresentation=AMSPublic&PosseObjectDef=o_ATIS_DocumentSearch&PosseMenuName=WS_Main&Criteria_LicenceNumber=501526"
                        target="_blank"
                      >
                        
                  501526
                
                      </a>
                    </td>
                    <td
                      aria-colindex="2"
                      class=""
                      role="cell"
                    >
                      <ul
                        class="p-0 m-0"
                      >
                        <li>
                          <router-link-stub
                            to="[object Object]"
                          >
                            117043
                          </router-link-stub>
                        </li>
                      </ul>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      class=""
                      role="cell"
                    >
                      <a
                        href="https://j200.gov.bc.ca/pub/ams/Default.aspx?PossePresentation=AMSPublic&PosseObjectDef=o_ATIS_DocumentSearch&PosseMenuName=WS_Main&Criteria_LicenceNumber=501527"
                        target="_blank"
                      >
                        
                  501527
                
                      </a>
                    </td>
                    <td
                      aria-colindex="2"
                      class=""
                      role="cell"
                    >
                      <ul
                        class="p-0 m-0"
                      >
                        <li>
                          <router-link-stub
                            to="[object Object]"
                          >
                            117044
                          </router-link-stub>
                        </li>
                      </ul>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      class=""
                      role="cell"
                    >
                      <a
                        href="https://j200.gov.bc.ca/pub/ams/Default.aspx?PossePresentation=AMSPublic&PosseObjectDef=o_ATIS_DocumentSearch&PosseMenuName=WS_Main&Criteria_LicenceNumber=501528"
                        target="_blank"
                      >
                        
                  501528
                
                      </a>
                    </td>
                    <td
                      aria-colindex="2"
                      class=""
                      role="cell"
                    >
                      <ul
                        class="p-0 m-0"
                      >
                        <li>
                          <router-link-stub
                            to="[object Object]"
                          >
                            117045
                          </router-link-stub>
                        </li>
                      </ul>
                    </td>
                  </tr>
                  <!---->
                  <!---->
                </tbody>
              </table>
               
              <p>
                <i>
                  Licence info last updated 11/4/2019
                </i>
              </p>
               
              <p>
                
              Licensing information is obtained from
              the 
                <a
                  class="d-print-url"
                  href="https://catalogue.data.gov.bc.ca/dataset/water-rights-licences-public"
                  target="_blank"
                >
                  
                Water Rights Licence - Public data layer
              
                </a>
                .
            
              </p>
               
              <p>
                
              Unique licenses are counted once for each aquifer that they are associated with.
            
              </p>
               
              <p>
                
              The total licensed volume is counted once for each licence (the total volume may
              be shared between wells if there are multiple wells in a licence). In cases where
              specific volumes are licensed for multiple purposes, individual volumes are summed.
            
              </p>
            </div>
             
            <div
              class="knowledge-indicators col-lg-6 col-xl-4 col-12"
            >
              <h5
                class="mt-3 border-bottom pb-4 main-title"
              >
                Knowledge Indicators
              </h5>
               
              <ul
                class="ml-0 mr-0 mb-0 mt-4 p-0 aquifer-information-list"
              >
                <div
                  class="aquifer-information-list-divider"
                />
                 
                <li>
                  <div
                    class="advanced-mapping"
                  >
                    <dt
                      class="text-right"
                    >
                      Advanced mapping
                    
                      <i
                        class="fa fa-question-circle color-info fa-xs pt-0 mt-0 d-print-none"
                        id="aquifer-advanced-mapping"
                        tabindex="0"
                      />
                       
                      <b-popover-stub
                        content="Aquifers with advanced mapping have been mapped in three dimensions. Generally, cross-sections and/or three-dimensional models, have been developed."
                        target="aquifer-advanced-mapping"
                        triggers="hover focus"
                      />
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Artesian advisory
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div
                    class="observational-wells"
                  >
                    <dt
                      class="text-right"
                    >
                      Observation wells
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                      <div>
                        <h6
                          class="border-bottom"
                        >
                          Active
                        </h6>
                         
                        <ul
                          class="p-0 m-0"
                        >
                          <li
                            class=""
                          >
                            <span
                              class="observation-well"
                            >
                              <a
                                class="d-print-url"
                                href="https://governmentofbc.maps.arcgis.com/apps/webappviewer/index.html?id=b53cb0bf3f6848e79d66ffd09b74f00d&find=OBS WELL 2"
                                target="_blank"
                              >
                                
    2
  
                              </a>
                               
                              <!---->
                               
                              <!---->
                            </span>
                          </li>
                          <li
                            class=""
                          >
                            <span
                              class="observation-well"
                            >
                              <a
                                class="d-print-url"
                                href="https://governmentofbc.maps.arcgis.com/apps/webappviewer/index.html?id=b53cb0bf3f6848e79d66ffd09b74f00d&find=OBS WELL 2"
                                target="_blank"
                              >
                                
    2
  
                              </a>
                               
                              <!---->
                               
                              <!---->
                            </span>
                          </li>
                        </ul>
                      </div>
                       
                      <!---->
                       
                      <!---->
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Numerical model
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Pumping stress index
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Water budget
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div
                    class="water-quality-information"
                  >
                    <dt
                      class="text-right"
                    >
                      Water quality information
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                      <router-link-stub
                        to="[object Object]"
                      >
                        
                      3 wells with an EMS ID
                    
                      </router-link-stub>
                    </dd>
                  </div>
                </li>
                <li>
                  <div
                    class="aquifer-connected"
                  >
                    <dt
                      class="text-right"
                    >
                      Hydraulically connected (screening level)
                    
                      <i
                        class="fa fa-question-circle color-info fa-xs pt-0 mt-0 d-print-none"
                        id="aquiferConnectedInfo"
                        tabindex="0"
                      />
                       
                      <b-popover-stub
                        content="Inferred based on aquifer subtype - not field verified."
                        target="aquiferConnectedInfo"
                        triggers="hover focus"
                      />
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                      More likely
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Groundwater Surface Water Interactions
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
                <li>
                  <div>
                    <dt
                      class="text-right"
                    >
                      Other information
                    </dt>
                     
                    <dd
                      class="m-0"
                    >
                       
                      <p
                        class="m-0"
                      >
                        No information available.
                      </p>
                    </dd>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
